### PR TITLE
Test updates

### PR DESCRIPTION
**Pull request by avaris**
* Adjust namespace plugin tests to account for already existing namespace plugins
* Adds a test case for #2488. One draft with a date and one without a date triggers the issue.

### CODE REVIEW

File: pelican/tests/test_plugins.py [hunk 1]
```diff
@@ -66,35 +66,38 @@ def test_namespace_path_modification(self):
         self.assertEqual(pelican.__path__, old_path)
 
     def test_get_namespace_plugins(self):
-        # without plugins
-        ns_plugins = get_namespace_plugins()
-        self.assertEqual(len(ns_plugins), 0)
+        # existing nanmespace plugins
```
Comment (justinmayer) [inline pelican/tests/test_plugins.py:69]: nanmespace → namespace

File: pelican/tests/test_plugins.py [hunk 2]
```diff
@@ -66,35 +66,38 @@ def test_namespace_path_modification(self):
...
     def test_load_plugins(self):
-        # no plugins
-        plugins = load_plugins({})
-        self.assertEqual(len(plugins), 0)
+        def get_plugin_names(plugins):
+            return set(
+                plugin.NAME if hasattr(plugin, 'NAME') else plugin.__name__
+                for plugin in plugins)
+
+        # existing nanmespace plugins
```
Comment (justinmayer) [inline pelican/tests/test_plugins.py:91]: nanmespace → namespace

Comment (avaris) [reply pelican/tests/test_plugins.py]: :+1: done

File: pelican/tests/test_plugins.py (hunk 1 above)
Comment (avaris) [reply pelican/tests/test_plugins.py]: :+1: done

**Comment by justinmayer** [comment]
Many thanks for the test enhancements, @avaris! ✨

**Pull request merged**